harness/determined/pytorch/dsat/_dsat_search_method.py

```diff
@@ -284,7 +294,7 @@ def enforce_consistent_batch_config(self, hparams: Dict[str, Any]) -> None:
     def update_trial_metric(
         self,
         trial: DSATTrial,
-        metric: Dict[str, Any],
+        metric: Union[float, Dict[str, Any]],
```

Comment:
The actual `metric` should always be a `Dict[str, Any]` for DS AT. Is this typing needed for consistency with the base `SearchMethod` class's methods or something?

Reply:
Basically, yes. Agreed, and there probably is a way to make the types play nice, but I didn't find it yet